### PR TITLE
Added login/logout functionality. Made changes to the Course and Temp…

### PR DESCRIPTION
**Pull request by lauradobson**
…Course model to account for no null values in all fields except instructor name and instructor email.

### CODE REVIEW

File: taworks/taform/models.py [hunk 1]
```diff
@@ -33,19 +33,19 @@ class Student(models.Model):
 
 class Course(models.Model):
     term = models.PositiveIntegerField(null=True, validators=[MaxValueValidator(9999), MinValueValidator(1000)])
-    course_subject = models.CharField(max_length=255)
-    course_id = models.CharField(max_length=255)
-    section = models.CharField(max_length=255)
-    course_name = models.CharField(max_length=255)
+    course_subject = models.CharField(null=True, max_length=255)
```
Comment (j2kan) [inline taworks/taform/models.py]: Can you move all the null=True to come after max_length just to be consistent with the rest of the models?

File: taworks/taform/models.py [hunk 2]
```diff
@@ -33,19 +33,19 @@ class Student(models.Model):
...
 class TempCourse(models.Model):
     term = models.PositiveIntegerField(null=True, validators=[MaxValueValidator(9999), MinValueValidator(1000)])
-    course_subject = models.CharField(max_length=255)
-    course_id = models.CharField(max_length=255)
-    section = models.CharField(max_length=255)
-    course_name = models.CharField(max_length=255)
+    course_subject = models.CharField(null=True, max_length=255)
+    course_id = models.CharField(null=True, max_length=255)
+    section = models.CharField(null=True, max_length=255)
+    course_name = models.CharField(null=True, max_length=255)
```
Comment (j2kan) [inline taworks/taform/models.py]: same as above.

File: taworks/taform/templates/taform/course_list.html [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
-
-<form action="{% url 'course_list' %}" method="post" enctype="multipart/form-data">
-{% csrf_token %}
-    <input type="file" name="csv_file" />
-    <input type="submit" name="Upload" value="Upload"/>
-</form>
-
+{% if user.is_authenticated %}
+    <form>
+            <button class="btn-lg btn-success" formaction="{% url 'home' %}" style="vertical-align:top; float: right">Home</button>
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: add css to static/taform/css/style.css

Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: check spacing around this block

Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: add {% csrf_token %}

File: taworks/taform/templates/taform/course_list.html [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
+        </form>
+    <div>
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: remove indent

File: taworks/taform/templates/taform/course_list.html [hunk 5]
```diff
@@ -1,13 +1,17 @@
...
+        <h1>Upload Courses</h1>
+    </div>
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: remove indent

File: taworks/taform/templates/taform/course_list.html [hunk 6]
```diff
@@ -1,13 +1,17 @@
...
+    <font color="red">{{error}}</font>
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: add css to static/taform/css/style.css

fix spacing int his area

File: taworks/taform/templates/taform/course_success.html [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{% if user.is_authenticated %}
+    <form>
+            <button class="btn-lg btn-success" formaction="{% url 'home' %}" style="vertical-align:top; float: right">Home</button>
+        </form>
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_success.html]: fix indent

File: taworks/taform/templates/taform/course_success.html [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{% if user.is_authenticated %}
+    <form>
+            <button class="btn-lg btn-success" formaction="{% url 'home' %}" style="vertical-align:top; float: right">Home</button>
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_success.html]: add css to static/taform/css/style.css

Comment (j2kan) [inline taworks/taform/templates/taform/course_success.html]: add cross site forgery token

File: taworks/taform/templates/taform/home.html [hunk 9]
```diff
@@ -0,0 +1,38 @@
+{% if user.is_authenticated %}
+{% load staticfiles %}
+
+    <html>
```
Comment (j2kan) [inline taworks/taform/templates/taform/home.html]: remove indent, don't need to indent at start of file

File: taworks/taform/templates/taform/home.html [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    <head>
+        <link rel="stylesheet" type="text/css" href="{% static 'taform/css/style.css' %}" />
+        <title>Home</title>
+    </head>
+    <body>
+        
+        <form>
```
Comment (j2kan) [inline taworks/taform/templates/taform/home.html]: remove indent, 
add css to static/taform/css/style.css
add CSRF.token

File: taworks/taform/templates/taform/home.html [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+            <button class="btn-lg btn-success" formaction="{% url 'logout' %}" style="vertical-align:top; float: right">Logout</button>
+        </form>
+        
+        <h1><legend>TA Application Home Page</legend></h1>
```
Comment (j2kan) [inline taworks/taform/templates/taform/home.html]: too much indent

File: taworks/taform/templates/taform/test.html [hunk 12]
```diff
@@ -1,4 +1,10 @@
-<div>
-    <h1>Success!</h1>
-</div>
-
+{% if user.is_authenticated %}
+    <form>
```
Comment (j2kan) [inline taworks/taform/templates/taform/test.html]: remove indent

File: taworks/taform/templates/taform/test.html [hunk 13]
```diff
@@ -1,4 +1,10 @@
...
+            <button class="btn-lg btn-success" formaction="{% url 'home' %}" style="vertical-align:top; float: right">Home</button>
```
Comment (j2kan) [inline taworks/taform/templates/taform/test.html]: add css to static/taform/css/style.css

File: taworks/taform/urls.py [hunk 14]
```diff
@@ -6,8 +6,11 @@
 
 urlpatterns = [
     url(r'^$', views.apply, name='apply'),
+    url(r'application.html', views.apply, name='application'),
```
Comment (j2kan) [inline taworks/taform/urls.py:8]: we can remove this and change line 30 from home.html from
<td class="tg-yw4l"><a href="application.html">TA Application</a></td>
to
<td class="tg-yw4l"><a href="/">TA Application</a></td>

File: taworks/taform/urls.py [hunk 15]
```diff
@@ -6,8 +6,11 @@
...
     url(r'application_submitted.html', views.application_submitted, name='application_submitted'),
     url(r'course_list.html', views.course_list, name='course_list'),
-
+    url(r'home.html', views.home, name='home'),
+    url(r'^logout.html', views.logout, name='logout'),
```
Comment (j2kan) [inline taworks/taform/urls.py]: remove ^ on line 13 and 14, consistent with the rest

File: taworks/taform/views.py [hunk 16]
```diff
@@ -10,6 +10,14 @@
 from django.template import loader
 import csv
 import codecs
+from django.contrib.auth import logout as django_logout
+
+def home(request):
+    return render(request, 'taform/home.html')
+
+def logout(request):
+    #logout(request)
```
Comment (j2kan) [inline taworks/taform/views.py]: remove

File: taworks/taform/views.py [hunk 17]
```diff
@@ -47,22 +55,27 @@ def application_submitted(request):
...
         models.TempCourse.objects.all().delete()
         f = request.FILES['csv_file']
-        SaveTemp(f)
-        courses = models.TempCourse.objects.all()
-        template = loader.get_template("taform/confirmation.html")
-        return render(request, 'taform/confirmation.html', {'courses': courses})
+        try:
+            save_temp(f)
+            courses = models.TempCourse.objects.all()
+            template = loader.get_template("taform/confirmation.html")
```
Comment (j2kan) [inline taworks/taform/views.py]: switch to single quotes like the rest to be consistent

File: taworks/taform/views.py [hunk 18]
```diff
@@ -74,11 +87,27 @@ def SaveTemp(f):
...
+    import mimetypes
+
+    filename     = "../../static/taform/course_template.csv" # Select your file here.
+    download_name ="Course Template.csv"
+    wrapper      = FileWrapper(open(filename))
+    content_type = mimetypes.guess_type(filename)[0]
+    response     = HttpResponse(wrapper,content_type=content_type)
+    response['Content-Length']      = os.path.getsize(filename)    
+    response['Content-Disposition'] = "attachment; filename=Course Template.csv"
+     #%s"%download_name
```
Comment (j2kan) [inline taworks/taform/views.py]: remove

File: taworks/taform/views.py [hunk 19]
```diff
@@ -74,11 +87,27 @@ def SaveTemp(f):
...
     newobjects = [models.Course(**values) for values in queryset]
     models.Course.objects.bulk_create(newobjects)
 
+def send_file(request):
+    import os, tempfile, zipfile
+    from django.core.servers.basehttp import FileWrapper
+    from django.conf import settings
+    import mimetypes
+
+    filename     = "../../static/taform/course_template.csv" # Select your file here.
```
Comment (j2kan) [inline taworks/taform/views.py]: remove spaces for all the equal signs

File: taworks/taform/templates/taform/test.html [hunk 20]
```diff
@@ -1,4 +1,10 @@
-<div>
-    <h1>Success!</h1>
-</div>
-
+{% if user.is_authenticated %}
```
Comment (j2kan) [inline taworks/taform/templates/taform/test.html]: Please rename this file to something like sucess.html

**Review by j2kan — CHANGES_REQUESTED** [review]
great lot of good features.

notice that 
style="vertical-align:top; float: right">Home</button>
was used in multiple places.

The DRY principal is a good reason why we should move these to the static/css/ folder.

File: taworks/taform/urls.py (hunk 14 above)
Comment (lauradobson) [reply taworks/taform/urls.py:8]: explain?

File: taworks/taform/templates/taform/course_list.html (hunk 3 above)
Comment (lauradobson) [inline taworks/taform/templates/taform/course_list.html]: it's already in there?

Comment (lauradobson) [reply taworks/taform/templates/taform/course_list.html]: it's already in there?

Comment (j2kan) [reply taworks/taform/templates/taform/course_list.html]: I saw  <form> tags and didn't see {% csrf_token %}
you should remove the  <form> tags that isn't being used all together.

File: taworks/taform/static/taform/css/style.css [hunk 21]
```diff
@@ -6,4 +6,9 @@ table{}
 .tg td{font-family:Arial, sans-serif;font-size:14px;padding:10px 5px;border-style:solid;border-width:0px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#fff;border-top-width:1px;border-bottom-width:1px;}
```
Comment (j2kan) [inline taworks/taform/static/taform/css/style.css]: can you make the style sheet follow the same style.

i.e.

.someDiv {
  background: red;
  padding: 2em;
  border: 1px solid black;
}

File: taworks/taform/templates/taform/course_list.html [hunk 22]
```diff
@@ -1,13 +1,19 @@
+{% load staticfiles %}
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: this should be changed to load static

File: taworks/taform/templates/taform/course_list.html [hunk 23]
```diff
@@ -1,13 +1,19 @@
...
+    <button class="button" formaction="{% url 'home' %}">Home</button>
+</form>
 <div>
     <h1>Upload Courses</h1>
 </div>
-
-<p>Please ensure the file you upload is in CSV format. This file must contain the course ID, name, instructor and email information. Click <a href="CourseTemplate.csv" download="CourseTemplate.csv">here</a> for the CSV template. Please note this will remove existing data.
-
-
+<font color="red">{{error}}</font>
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: can you add to style.css for color="red"? I can see it being used in other places as well for other pages.

File: taworks/taform/templates/taform/course_list.html [hunk 24]
```diff
@@ -1,13 +1,19 @@
+{% load staticfiles %}
+<link rel="stylesheet" type="text/css" href="{% static 'taform/css/style.css' %}" />
+{% if user.is_authenticated %}
+<form>
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: indent the block after {% if user.is_authenticated %}

File: taworks/taform/templates/taform/course_success.html [hunk 25]
```diff
@@ -0,0 +1,13 @@
...
+{% csrf_token %}
+{% if user.is_authenticated %}
+<form>
+    <button class="button" formaction="{% url 'home' %}">Home</button>
+</form>
+<div>
+    <h1>Courses successfully added to the database.</h1>
+</div>
+{% else %}
+<a href="{% url 'login' %}">login</a>
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_success.html:13]: indent this line

File: taworks/taform/templates/taform/home.html [hunk 26]
```diff
@@ -0,0 +1,38 @@
+{% if user.is_authenticated %}
+{% load staticfiles %}
+{% csrf_token %}
```
Comment (j2kan) [inline taworks/taform/templates/taform/home.html]: this should only be used inside form ?

File: taworks/taform/templates/taform/course_list.html [hunk 27]
```diff
@@ -1,13 +1,19 @@
...
+    <button class="button" formaction="{% url 'home' %}">Home</button>
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: Is this button for logout?

File: taworks/taform/views.py [hunk 28]
```diff
@@ -47,22 +54,31 @@ def application_submitted(request):
     return render(request, 'taform/application_submitted.html')
 
 def course_list(request):
```
Comment (j2kan) [inline taworks/taform/views.py]: Do we need @login_required here?

https://docs.djangoproject.com/en/1.11/topics/auth/default/

File: taworks/taform/templates/taform/home.html [hunk 29]
```diff
@@ -0,0 +1,38 @@
+{% if user.is_authenticated %}
```
Comment (j2kan) [inline taworks/taform/templates/taform/home.html]: Can you do the user.is_authenticated check as part of views.py?

for example:
def home(request):
    if not request.user.is_authenticated:
        return render(request, 'myapp/login_error.html')
    else:
        return render(request, 'taform/home.html')

File: taworks/taform/templates/taform/home.html [hunk 30]
```diff
@@ -0,0 +1,38 @@
...
+{% csrf_token %}
+<html>
+<head>
+    <link rel="stylesheet" type="text/css" href="{% static 'taform/css/style.css' %}" />
+    <title>Home</title>
+</head>
+<body>
+    
+    <form>
+        <button class="button" formaction="{% url 'logout' %}">Logout</button>
```
Comment (j2kan) [inline taworks/taform/templates/taform/home.html]: does it work with csrf_token added here?

File: taworks/taform/urls.py (hunk 14 above)
Comment (j2kan) [reply taworks/taform/urls.py:8]: not sure what that was, was pretty late when i reviewed

File: taworks/taform/views.py [hunk 31]
```diff
@@ -73,12 +89,33 @@ def SaveTemp(f):
         tmp.section = line[3]
         tmp.course_name = line[4]
         tmp.instructor_name = line[5]
-        tmp.instructor_email = line[6]
+        tmp.instructor_email = line[6]  
         tmp.save()
 
-def CopyCourses(newtable, oldtable):
+def validate_temp():
```
Comment (j2kan) [inline taworks/taform/views.py]: very nice udf for validating.

File: taworks/taform/views.py [hunk 32]
```diff
@@ -73,12 +89,33 @@ def SaveTemp(f):
...
+    courses = models.TempCourse.objects.all()
+    for course in courses:
+        if not course.term or course.term<1000 or course.term>9999 or not course.course_id or not course.course_subject or not course.section or not course.course_name:
+            return False
+    return True
```
Comment (j2kan) [inline taworks/taform/views.py]: If there was an issue inserting into TempCourse table, this function will return True.
I think the function should return False if there was nothing to validate in the first place.

File: taworks/taform/views.py [hunk 33]
```diff
@@ -47,22 +54,31 @@ def application_submitted(request):
...
-        SaveTemp(f)
+        try:
+            save_temp(f)
+        except:
+            return render(request, 'taform/course_list.html', {'error': 'There is an error with the CSV file. Please refer to the template and try again.'})   
+        is_valid = validate_temp()  
         courses = models.TempCourse.objects.all()
-        template = loader.get_template("taform/confirmation.html")
-        return render(request, 'taform/confirmation.html', {'courses': courses})
+        if is_valid == True:
```
Comment (j2kan) [inline taworks/taform/views.py]: nit.
you can just have if is_valid:

File: taworks/taform/views.py [hunk 34]
```diff
@@ -47,22 +54,31 @@ def application_submitted(request):
...
+            template = loader.get_template('taform/confirmation.html')
```
Comment (j2kan) [inline taworks/taform/views.py]: what does template do here?

File: taworks/taform/views.py [hunk 35]
```diff
@@ -73,12 +89,33 @@ def SaveTemp(f):
...
+            return False
+    return True
+
+def copy_courses(newtable, oldtable):
     models.Course.objects.all().delete()
     queryset = models.TempCourse.objects.all().values('term', 'course_subject', 'course_id', 'section', 'course_name', 'instructor_name', 'instructor_email')
     newobjects = [models.Course(**values) for values in queryset]
     models.Course.objects.bulk_create(newobjects)
 
+def send_file(request):
```
Comment (j2kan) [inline taworks/taform/views.py:123]: Can you move this above your user defined function?

File: taworks/taform/views.py [hunk 36]
```diff
@@ -47,22 +54,31 @@ def application_submitted(request):
...
+        is_valid = validate_temp()  
         courses = models.TempCourse.objects.all()
-        template = loader.get_template("taform/confirmation.html")
-        return render(request, 'taform/confirmation.html', {'courses': courses})
+        if is_valid == True:
+            template = loader.get_template('taform/confirmation.html')
+            return render(request, 'taform/confirmation.html', {'courses': courses})
+        else:
+            return render(request, 'taform/course_list.html', {'error': 'There is an error with the CSV file. Please refer to the template and try again.'})   
     if 'Submit' in request.POST:
```
Comment (j2kan) [inline taworks/taform/views.py]: Does the submit and cancel button only show up after csv upload?

File: taworks/taform/views.py [hunk 37]
```diff
@@ -73,12 +89,33 @@ def SaveTemp(f):
...
+    return True
+
+def copy_courses(newtable, oldtable):
     models.Course.objects.all().delete()
     queryset = models.TempCourse.objects.all().values('term', 'course_subject', 'course_id', 'section', 'course_name', 'instructor_name', 'instructor_email')
     newobjects = [models.Course(**values) for values in queryset]
     models.Course.objects.bulk_create(newobjects)
 
+def send_file(request):
+    import os, tempfile, zipfile
```
Comment (j2kan) [inline taworks/taform/views.py]: move these import statements to the top of this file.

**Review by j2kan — CHANGES_REQUESTED** [review]

File: taworks/taform/templates/taform/course_list.html (hunk 27 above)
Comment (lauradobson) [reply taworks/taform/templates/taform/course_list.html]: Nope, its to return back to the home page from the Upload Courses page

File: taworks/taform/templates/taform/course_success.html [hunk 38]
```diff
@@ -0,0 +1,13 @@
+{% load staticfiles %}
+<link rel="stylesheet" type="text/css" href="{% static 'taform/css/style.css' %}" />
+{% csrf_token %}
+{% if user.is_authenticated %}
+<form>
```
Comment (lauradobson) [inline taworks/taform/templates/taform/course_success.html]: Sighhh you told me this didn't need to be indented so I removed it

Comment (lauradobson) [reply taworks/taform/templates/taform/course_success.html]: Will add back

File: taworks/taform/templates/taform/home.html (hunk 29 above)
Comment (lauradobson) [reply taworks/taform/templates/taform/home.html]: I can try but imagine this as php sessions.

File: taworks/taform/views.py (hunk 36 above)
Comment (lauradobson) [reply taworks/taform/views.py]: yep..... it's on the confirmation page.

File: taworks/taform/views.py (hunk 32 above)
Comment (lauradobson) [reply taworks/taform/views.py]: I don't understand how that will make a difference. Regardless, the form is being validated

File: taworks/taform/views.py (hunk 28 above)
Comment (lauradobson) [reply taworks/taform/views.py]: no it's working w/o it

File: taworks/taform/views.py (hunk 37 above)
Comment (lauradobson) [reply taworks/taform/views.py]: from django.core.servers.basehttp import FileWrapper needs to be inside the function or else it throws and error

File: taworks/taform/static/taform/css/style.css [hunk 39]
```diff
@@ -1,9 +1,59 @@
...
-table{}
-.tg  {border-collapse:collapse;border-spacing:0;border-color:#ccc;}
-.tg td{font-family:Arial, sans-serif;font-size:14px;padding:10px 5px;border-style:solid;border-width:0px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#fff;border-top-width:1px;border-bottom-width:1px;}
-.tg th{font-family:Arial, sans-serif;font-size:14px;font-weight:normal;padding:10px 5px;border-style:solid;border-width:0px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#f0f0f0;border-top-width:1px;border-bottom-width:1px;}
-.tg .tg-8fvv{font-size:14px;background-color:#9b9b9b;color:#ffffff;vertical-align:top;text-align: left;}
-.tg .tg-yw4l{vertical-align:top}
\ No newline at end of file
+
+#font{
+    color: #FE0909;
```
Comment (j2kan) [inline taworks/taform/static/taform/css/style.css:6]: lots of white space showing up in the git hub review

File: taworks/taform/static/taform/css/style.css [hunk 40]
```diff
@@ -1,9 +1,59 @@
...
+}
+
+table {
```
Comment (j2kan) [inline taworks/taform/static/taform/css/style.css:9]: remove?

File: taworks/taform/static/taform/css/style.css [hunk 41]
```diff
@@ -1,9 +1,59 @@
...
+    vertical-align: top;
+    text-align: left;
+}
+.tg .tg-yw4l {
+    vertical-align: top
+}
+.button {
+    vertical-align: top;
+    float: right;
+}
```
Comment (j2kan) [inline taworks/taform/static/taform/css/style.css:59]: looking really good!!!

File: taworks/taform/templates/taform/confirmation.html [hunk 42]
```diff
@@ -1,10 +1,11 @@
-{% load staticfiles %}
+{% load static %}
 <link rel="stylesheet" type="text/css" href="{% static 'taform/css/style.css' %}" />
 <div>
     <h1>Upload Courses</h1>
 </div>
 
-<p>You would like to upload the following data. Is this correct? <font color="red">Submitting this data will reset the database for the upcoming term and all existing applications will be deleted. </font>If yes, select "Commit". If no, select "Don't Commit".
+<p>You would like to upload the following data. Is this correct? If yes, select "Commit". If no, select "Don't Commit".
```
Comment (j2kan) [inline taworks/taform/templates/taform/confirmation.html]: ending </p>?

Comment (j2kan) [reply taworks/taform/templates/taform/confirmation.html]: uhh it skipped it ending paragraph tag

File: taworks/taform/templates/taform/confirmation.html [hunk 43]
```diff
@@ -1,10 +1,11 @@
-{% load staticfiles %}
+{% load static %}
 <link rel="stylesheet" type="text/css" href="{% static 'taform/css/style.css' %}" />
 <div>
     <h1>Upload Courses</h1>
 </div>
 
-<p>You would like to upload the following data. Is this correct? <font color="red">Submitting this data will reset the database for the upcoming term and all existing applications will be deleted. </font>If yes, select "Commit". If no, select "Don't Commit".
+<p>You would like to upload the following data. Is this correct? If yes, select "Commit". If no, select "Don't Commit". 
+<div id="font">Please note: Submitting this data will reset the database for the upcoming term and all existing applications will be deleted. </div>
```
Comment (j2kan) [inline taworks/taform/templates/taform/confirmation.html]: to be consistent
div
     text
div

File: taworks/taform/templates/taform/course_list.html [hunk 44]
```diff
@@ -1,10 +1,14 @@
...
+</form>
 <div>
     <h1>Upload Courses</h1>
 </div>
-
-<p>Please ensure the file you upload is in CSV format. This file must contain the course ID, name, instructor and email information. Click <a href="CourseTemplate.csv" download="CourseTemplate.csv">here</a> for the CSV template. Please note this will remove existing data.
-
-
+<div id="font">{{error}}</div>
+<p>Please ensure the file you upload is in CSV format. This file must contain the course ID, name, instructor and email information. Click <a href="../../static/taform/course_template.csv" download="course_template.csv">here</a> for the CSV template. Please note this will remove existing data.
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_list.html]: ending p tag?
also i think with p tag you can just press enter and have this over multiple lines right?

File: taworks/taform/templates/taform/course_success.html [hunk 45]
```diff
@@ -0,0 +1,13 @@
+{% load static %}
+<link rel="stylesheet" type="text/css" href="{% static 'taform/css/style.css' %}" />
+{% csrf_token %}
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_success.html]: this should be after the form tag

File: taworks/taform/templates/taform/home.html [hunk 46]
```diff
@@ -0,0 +1,35 @@
+{% load static %}
+
+<html>
+<head>
+    <link rel="stylesheet" type="text/css" href="{% static 'taform/css/style.css' %}" />
+    <title>Home</title>
+</head>
+<body>
+
```
Comment (j2kan) [inline taworks/taform/templates/taform/home.html:9]: remove extra line

**Review by j2kan — CHANGES_REQUESTED** [review]

File: taworks/taform/templates/taform/confirmation.html (hunk 43 above)
Comment (lauradobson) [reply taworks/taform/templates/taform/confirmation.html]: ill just take the heading out of a div...

File: taworks/taform/static/taform/css/style.css (hunk 40 above)
Comment (lauradobson) [reply taworks/taform/static/taform/css/style.css:9]: nope this is needed for the rest of the custom css tags

File: taworks/taform/static/taform/css/style.css (hunk 39 above)
Comment (lauradobson) [reply taworks/taform/static/taform/css/style.css:6]: not sure why..isn't this way in sublime (did the tab thing to make sure)

File: taworks/taform/static/taform/css/style.css (hunk 40 above)
Comment (j2kan) [reply taworks/taform/static/taform/css/style.css:9]: ok

File: taworks/taform/static/taform/css/style.css (hunk 39 above)
Comment (j2kan) [reply taworks/taform/static/taform/css/style.css:6]: yeah looks good now

File: taworks/taform/templates/taform/confirmation.html [hunk 47]
```diff
@@ -1,10 +1,11 @@
-{% load staticfiles %}
+{% load static %}
 <link rel="stylesheet" type="text/css" href="{% static 'taform/css/style.css' %}" />
-<div>
-    <h1>Upload Courses</h1>
-</div>
 
-<p>You would like to upload the following data. Is this correct? <font color="red">Submitting this data will reset the database for the upcoming term and all existing applications will be deleted. </font>If yes, select "Commit". If no, select "Don't Commit".
+<h1>Upload Courses</h1>
+
```
Comment (j2kan) [inline taworks/taform/templates/taform/confirmation.html:5]: remove extra line

File: taworks/taform/templates/taform/course_success.html [hunk 48]
```diff
@@ -0,0 +1,14 @@
+{% load static %}
+<link rel="stylesheet" type="text/css" href="{% static 'taform/css/style.css' %}" />
+
+{% if user.is_authenticated %}
+<form>
+	{% csrf_token %}
```
Comment (j2kan) [inline taworks/taform/templates/taform/course_success.html:6]: seems too indented

**Review by j2kan — APPROVED** [review]

**Comment by j2kan** [comment]
👍

**Pull request merged**